src/tools/tidy/src/deps.rs

```diff
@@ -148,6 +152,7 @@ const WHITELIST: &[Crate<'_>] = &[
     Crate("serde"),
     Crate("serde_derive"),
     Crate("smallvec"),
+    Crate("spin"),
```

Comment:
I think this is something we should wait for a new publish of `getrandom` for unfortunately. This makes it so all of `lazy_static` users spin waiting for initialization, which isn't appropriate for the compiler's use case where we're not interested in burning more CPU memory. I believe the master branch of `getrandom`, when published, should mean we do not need to add this crate.

Reply:
@josephlr do you think we could do a new `getrandom` release ? 

I think that I can just `cargo update -p getrandom` at that point and that should remove the dependency on lazy_static and the spin crate.

Reply:
I'm not a maintainer of the crate, so either @dhardy or @newpavlov would have to actually cut the release, but we should be able to cut `0.1.7` from current master. We have [some changes](https://github.com/rust-random/getrandom/pulls) in the works for https://github.com/rust-lang/rust/pull/62082, but none of it is critical for this. However, it will remove `spin`, which is the important part.

Updating will stop `getrandom` from using `lazy_static`, but not `rand` itself, due to a transitive dep, see my comment below.

Reply:
I hope we will be able to finish rust-random/getrandom#54 on this weekend, but ideally I also would like to merge rust-random/getrandom#58 (so I will not have to update `Cargo.lock` in #62082). But if you want I can release the current `master` branch as-is.

Reply:
FYI: we have published `getrandom v0.1.7` which does not depend on `lazy_static` anymore.

Reply:
Thanks! Please do let us know when the `ppv` crate removes the dependency as well!

Reply:
`ppv-lite86` has no dependencies, `c2-chacha` depends on `lazy_static` but it doesn't use `spin_no_std` feature.
In the other words when you update `getrandom` you will also drop `spin` dependecy and unblock this PR.